### PR TITLE
Update e2e tests to comply with Drawer Component

### LINKED ISSUE #98: Subtask: End-to-End test using `Object Selector` component
### Acceptance Criteria 
- Test loading via `uuid` in the address bar to navigate to an example `SSP`, `Catalog`, `Profile`, & `Component Definition`. 
- Test loading those same examples by using the `Drawer Selector`
  - `url` should also update to reflect the loaded object's `uuid`
  - Test that multiple of an `oscal object` are properly listed. For example, multiple `catalogs`. This may require adding some more example data. 
- Ensure that removing the `LoaderForm` while in `REST` mode, does not break functionality of tests.

### PR DESCRIPTION
**Pull request by schen2166**
Update e2e tests to comply with Drawer Component

Commands:
-Adjusted navToViewer function to comply with the new navigation
where the titleText parameter is removed because now navigating
using the drawer component sections.
-Adjusted the SSP, component, profile, and catalog Viewer
navigations to comply with the parameters used in the navtoViewer
function
- Removed navToTestSspRestMode function entirely
        -1) repeated the functionality found in navToViewer
        -2) the clickable box to labeled "Select Oscal SSP"
            will no longer be needed as it will be supported
            with the drawer component friendly URLs

Edit Tests:
-Replaced functions calls for navToTestSspRestMode with
navToSspViewer as an update

Load Tests:
-Replaced function calls for the navToTestSspRestMode with
navToSspViewer function call.
-Removed check for box click as it will be replaced by the drawer
component.

### CODE REVIEW

**Comment by kylelaker** [comment]
Git `user.email` needs to be fixed on this branch as well. Please work with @Bronstrom to correct this.

**Comment by kylelaker** [comment]
@Bronstrom @schen2166 The commits on this branch need to be amended and force-pushed to ensure they are linked correctly.

**Comment by schen2166** [comment]
The tests for this issue on GitHub failed. This may be because the default docker image is still routed to the `Docker image` without the drawer component. I've tested my cypress tests locally and they have all passed. 

**Note: The Catalog Tests occasionally fail** but that is because the Catalog is a large file and can take time to load. Currently, the test will wait 1 second for the Catalog to load (controlled by line 31 of `commands.js`). This can be resolved by increasing the cypress wait time or by rerunning the test.

The tests in Load and Edit don't pass because I've restructured the functions defined in `command.js` to accommodate the features implemented in the drawer. They pass in local testing as well.

File: end-to-end-tests/cypress/e2e/OSCAL/Drawer.cy.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    cy.navToCdefViewer(COMPONENT_NAVIGATION);
+    cy.contains("REST Mode").click();
+    cy.contains("Component");
+    cy.contains(COMPONENT_NAVIGATION);
+  });
+
+  it("Successfully Loads SSP in non REST", () => {
+    cy.navToSspViewer(SSP_NAVIGATION);
+    cy.contains("REST Mode").click();
+    cy.contains("SSP");
```
Comment (kylelaker) [inline end-to-end-tests/cypress/e2e/OSCAL/Drawer.cy.js:85]:
```suggestion
    cy.contains("System Security Plan");
```

**Review by kylelaker — CHANGES_REQUESTED** [review]

File: end-to-end-tests/cypress/e2e/OSCAL/OSCAL_Load_Tests.cy.js [hunk 2]
```diff
@@ -3,13 +3,13 @@ const COMP_DEF_TITLE_ORIG = 'Test Component Definition'
...
+    cy.navToSspViewer(SSP_TITLE_ORIG);
     cy.get(`[aria-label="show code"]`).click()
     cy.scrollTo('bottom')
     cy.contains('This is the control implementation for the system.').should('be.visible')
   })
   it('Successfully Loads SSPs by URL', () => {
-    cy.navToSspViewer()
+    cy.navToSspViewer(SSP_TITLE_ORIG)
     cy.contains("REST Mode").click()
     cy.contains('OSCAL SSP URL').first().should('exist').next().click().clear().type(Cypress.env('base_url') + "/oscal/v1/system-security-plans/cff8385f-108e-40a5-8f7a-82f3dc0eaba8")
```
Comment (kylelaker) [inline end-to-end-tests/cypress/e2e/OSCAL/OSCAL_Load_Tests.cy.js:14]:
```suggestion
    cy.contains('OSCAL System Security Plan UR:').first().should('exist').next().click().clear().type(Cypress.env('base_url') + "/oscal/v1/system-security-plans/cff8385f-108e-40a5-8f7a-82f3dc0eaba8")
```

**Comment by schen2166** [comment]
I looked into configuring the `defaultTimeout` and I'm wondering if that is really the best option to go with. Cypress's default timeout is 4000ms and if we increase that in the configuration file then that will increase the timeout for all commands regardless. 

The other approach to this is to change the timeout at a command level so that it will wait longer for that specific command for a longer time and leave the default time for the other commands to be 4000ms.

This is what I've been considering:
*Configuring Globally*:
If more files to be added in the future, then there might be additional wait time for the commands that don't take as much time to load and could potentially increase testing time where the tests are doing nothing. And the wait time is increased for this one component to load.
*Configuring at Command level*:
Makes the test run more efficient by allocating the exact time required for the commands that take longer to load but will be harder to maintain if there are multiple large files. Fortunately, defining your own macros in the config file can help with the maintenance by applying that macro only to those commands.

Which approach do you think is the best option?

**Comment by tuckerzp** [comment]
> Configuring at Command level

I think this is the best option.

File: end-to-end-tests/cypress/support/commands.js [hunk 3]
```diff
@@ -23,54 +23,57 @@
...
-  cy.visit(Cypress.env('base_url'))
-  cy.findByText('OSCAL Catalog Editor').should('exist')
-  cy.get('button').first().click()
-  cy.contains(viewerLinkText).click()
-  cy.findByText(titleText).should('be.visible')
-})
+Cypress.Commands.add("navToViewer", (viewerLinkText, navigationProfile) => {
+  cy.visit(Cypress.env("base_url"));
+  cy.get("button").first().click();
+  cy.contains(viewerLinkText).children('#class="MuiTreeItem-iconContainer"').click();
```
Comment (tuckerzp) [inline end-to-end-tests/cypress/support/commands.js:31]: Like here I feel like looking for a specific class of children is not really how a user would use the `Editor`. Why not just wait until the `Catalog` button is clickable, click it, then check for a specific file?

File: end-to-end-tests/cypress/e2e/OSCAL/Drawer.cy.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
+/*catalog Editor
+Profile Editor
+Component Editor
+System Security Plan Editor */
```
Comment (kylelaker) [inline end-to-end-tests/cypress/e2e/OSCAL/Drawer.cy.js:4]:
```suggestion
```

**Review by kylelaker — CHANGES_REQUESTED** [review]

File: .github/workflows/container-test-deploy.yml [hunk 5]
```diff
@@ -42,6 +42,8 @@ jobs:
           context: ./all-in-one
           load: true
           tags: ${{ env.IMAGE_NAME }}
+          build-args: |
+            OSCAL_REST_SERVICE_GIT_BRANCH=feature/combined-test
```
Comment (kylelaker) [inline .github/workflows/container-test-deploy.yml:46]: This must be reverted (and the related PRs must be merged) before this can be merged.

**Review by kylelaker — CHANGES_REQUESTED** [review]

**Review by kylelaker — APPROVED** [review]

**Pull request merged**